### ISSUE TITLE
Issues on Bravo

### ISSUE THREAD

**Issue by skeevy420** [Quarx2k/jordan-kernel#2]
Quarx, I flashed the Nov 04 Defy+ Custom Kernel rom on the Bravo (completely unedited).  Everything worked as expected (or didn't work as expected...camera).  The only real issue on the Bravo is the capacitive buttons aren't set properly -- they're mapped for the 4 Defy buttons.

While I know this isn't on your todo list, but would it be possible to add in Bravo camera support into the custom kernel?  Capacitive buttons are probably an easy fix (I've only looked at the keymaps for 5 minutes so far) and the only think keeping Bravo users from running the new Custom Kernel roms is the lack of camera support.  I'd have posted all this on xda, but the thread was locked when I when there this morning.  I haven't tried porting the camera libs like I normally do since I wanted to run the rom unmodified for better feedback.

**Comment by Quarx2k** [comment]
I compare defy and bravo driver, i'm not seen any differences in this part.
Defy froyo kernel vs bravo also same..
Maybe problem in devtree. or moto release wrong sources for bravo..

**Comment by skeevy420** [comment]
For the buttons or camera?  If buttons,then its acting like a Defy with the 4 buttons (more annoying than a bug since we only have the 3 buttons).  So far I've only built directly from  your sources (got a better pc; testing build environment) and haven't tried using Bravo devtree or anything like that yet -- actually, I recompiled with SoC camera enabled (didn't make Kobe Camera work -- still getting the 2 errors   W/WVMExtractor( 1210): Failed to open libwvm.so   &   I/Choreographer( 2270): Skipped 72 frames!  The application may be doing too much work on its main thread. 

 Everything else camera related looks like it normally does.  I'll do a build with Bravo devtree today and get back with you on it.

**Comment by Quarx2k** [comment]
Camera... can you try compile bravo kernel (from moto sources), use bravo devtree and  test camera?

**Comment by skeevy420** [comment]
Yeah, give me some time.  Need to recharge the battery :).  Bravo devtree with jordan kernel hangs at the 2nd-boot red led -- didn't reboot after 5 minutes -- can't provide any output or logs, no uart :(.  Tried using the dd of the devtree and another with all the padding removed; same results with both.  Hopefully whatever moto put in the kernel.tar.gz on sourceforge will work, fingers crossed.

UPDATE

compiling with kobe-kernel, replaced devtree in common, should be able to test it in an hour and a half

**Comment by skeevy420** [comment]
Defy kernel boots and works with Defy devtree -- acts like it has 4 capacitive buttons; camera\camcorder doesn't work; capacitive backlight always at 100%

Defy kernel hangs on 2nd-boot with Bravo devtree

Bravo kernel bootloops on 2nd-boot with Bravo devtree

Bravo kernel hangs on 2nd-boot with Defy devtree

I'm about to start pulling in commits from the 2.6 jordan branches onto the kobe sources.  Hopefully that'll get somewhere.

UPDATE

Replacing only /system/etc/mot_ise_imager_cfg.bin enabled the Bravo to record video (and play) in 848x480 -- 640x480 was the best until just now :).  

Logcat of the crash when taking a photo using Bravo .bin and libsoc --- http://forum.xda-developers.com/showpost.php?p=33830663&postcount=249
